numpyro/infer/mcmc.py

```diff
@@ -437,6 +444,8 @@ class HMC(MCMCKernel):
         value is :math:`2\\pi`.
     :param callable init_strategy: a per-site initialization function.
         See :ref:`init_strategy` section for available functions.
+    :param bool find_heuristic_step_size: whether to use an extra heuristic
```

Comment:
Maybe we can add to the description - `"Uses a heuristic function to adjust the step size at the end of each adaptation window."`?

Also, just to confirm - this will still use this to initialize the step size but omit doing so at the end of each adaptation window?

Reply:
@neerajprad We also skip the initialization of step size. Actually, this function adjusts step size at the "begin" of each adaptation window.

Reply:
I see, I forgot that this is a more fundamental change. I think the only time this should matter is when we are using very few warmup steps. It seems like we do this at the start and then at the end of each adaptation [window](https://github.com/pyro-ppl/numpyro/blob/master/numpyro/infer/hmc_util.py#L399)?

Reply:
> the only time this should matter is when we are using very few warmup steps

Exactly!

> It seems like we do this at the start and then at the end of each adaptation window?

Actually, we do this only at the start of each adaptation window (including the first window).